apis/src/websockets/api/schedules/handler.rs

```diff
@@ -0,0 +1,297 @@
+                            let game_ids: Vec<(Uuid, String)> = games::table
+                                .filter(games::tournament_id.eq(tournament.id))
+                                .filter(games::finished.eq(false))
+                                .filter(
+                                    games::white_id
+                                        .eq(self.user_id)
+                                        .or(games::black_id.eq(self.user_id)),
+                                )
+                                .select((games::id, games::nanoid))
+                                .get_results(tc)
+                                .await?;
+                            let mut schedules: HashMap<GameId, Vec<ScheduleResponse>> =
+                                HashMap::new();
+                            for (id, game_id) in game_ids {
+                                let current_schedules = game_schedules::table
+                                    .filter(game_schedules::game_id.eq(id))
+                                    .get_results(tc)
+                                    .await?;
+                                if !current_schedules.is_empty() {
+                                    for schedule in current_schedules {
+                                        let schedule: ScheduleResponse =
+                                            ScheduleResponse::from_model(
+                                                GameId(game_id.clone()),
+                                                &schedule,
+                                            )
+                                            .await?;
+                                        if let Some(v) = schedules.get_mut(&GameId(game_id.clone()))
+                                        {
+                                            v.push(schedule);
+                                        } else {
+                                            schedules
+                                                .insert(GameId(game_id.clone()), vec![schedule]);
+                                        }
+                                    }
+                                } else {
+                                    schedules.insert(GameId(game_id.clone()), vec![]);
+                                }
+                            }
+                            Ok(schedules)
+                        }
+                        .scope_boxed()
+                    })
+                    .await?;
+                messages.push(InternalServerMessage {
+                    destination: MessageDestination::User(self.user_id),
+                    message: ServerMessage::Schedule(ScheduleUpdate::OwnTournamentSchedules(
+                        schedules,
+                    )),
+                });
+            }
+        }
+        Ok(messages)
+    }
+    fn ensure_user_is_player(&self, game: &Game) -> Result<()> {
```

Comment:
This could maybe go into the request handler? So that we can share it with other handlers which do the same check. Not 100% sure though.